js/tests/unit/dropdown.spec.js

```diff
@@ -1783,4 +1783,105 @@ describe('Dropdown', () => {
+    document.addEventListener('click', delegatedClickListener)
+
+    btnDropdown.click()
+
+    expect(clickListener).toHaveBeenCalled()
+    expect(delegatedClickListener).toHaveBeenCalled()
+  })
+
+  it('should allow `.dropdown-menu form` click events to bubble up', () => {
+    fixtureEl.innerHTML = [
+      '<div class="dropdown">',
+      '  <button class="btn dropdown-toggle" data-bs-toggle="dropdown">Dropdown</button>',
+      '  <div class="dropdown-menu">',
+      '    <form>',
+      '      <input class="form-control" type="text">',
+      '    </form>',
+      '  </div>',
+      '</div>'
+    ].join('')
+
+    const btnDropdown = fixtureEl.querySelector('[data-bs-toggle="dropdown"]')
+    const form = fixtureEl.querySelector('form')
+    const formInput = form.querySelector('input')
+    const formClickListener = jasmine.createSpy('formClickListener')
+
+    form.addEventListener('click', formClickListener)
+
+    btnDropdown.addEventListener('shown.bs.dropdown', () => {
+      formInput.click()
+      form.click()
+    })
+
+    btnDropdown.addEventListener('hidden.bs.dropdown', () => {
+      throw new Error('should not throw hidden.bs.dropdown event')
+    })
+
+    btnDropdown.click()
+
+    expect(formClickListener).toHaveBeenCalledTimes(2)
+  })
+
+  it('should open the dropdown when clicking `data-bs-toggle="dropdown"`', done => {
+    fixtureEl.innerHTML = [
+      '<div class="dropdown">',
+      '  <button class="btn dropdown-toggle" data-bs-toggle="dropdown">Dropdown</button>',
+      '  <div class="dropdown-menu">',
+      '    <a class="dropdown-item" href="#subMenu">Sub menu</a>',
+      '  </div>',
+      '</div>'
+    ].join('')
+
+    const btnDropdown = fixtureEl.querySelector('[data-bs-toggle="dropdown"]')
+
+    btnDropdown.addEventListener('shown.bs.dropdown', () => setTimeout(() => {
+      expect(btnDropdown.classList.contains('show')).toEqual(true)
+      expect(btnDropdown.getAttribute('aria-expanded')).toEqual('true')
+      done()
+    }))
+
+    btnDropdown.click()
+  })
+
+  it('should open the dropdown when clicking the child element inside `data-bs-toggle="dropdown"`', done => {
+    fixtureEl.innerHTML = [
+      '<div class="container">',
+      '  <div class="dropdown">',
+      '    <button class="btn dropdown-toggle" data-bs-toggle="dropdown"><span id="childElement">Dropdown</span></button>',
+      '    <div class="dropdown-menu">',
+      '      <a class="dropdown-item" href="#subMenu">Sub menu</a>',
+      '    </div>',
+      '  </div>',
+      '</div>'
+    ].join('')
+
+    const btnDropdown = fixtureEl.querySelector('[data-bs-toggle="dropdown"]')
+    const childElement = fixtureEl.querySelector('#childElement')
+
+    btnDropdown.addEventListener('shown.bs.dropdown', () => setTimeout(() => {
+      expect(btnDropdown.classList.contains('show')).toEqual(true)
+      expect(btnDropdown.getAttribute('aria-expanded')).toEqual('true')
+      done()
+    }))
```

Comment:
`setTimeout` is not required here IMO

Reply:
The test wasn't working properly for me (locally) without the `setTimeout`.

Do you want me to remove all code related to checking for the form (with the idea being users should set `clickableMenu` instead of bootstrap detecting the presence of a form)?

Reply:
> Do you want me to remove all code related to checking for the form

No, that should be handled in the next PR.